src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -286,52 +476,119 @@ static size_t uct_cuda_copy_md_get_total_device_mem(CUdevice cuda_device)
     return 1; /* return 1 byte to avoid division by zero */
 }
 
+static ucs_status_t
+uct_cuda_copy_detect_vmm(const void *address, CUmemorytype *cuda_mem_type,
```

Comment:
1. make "address" void* instead of "const void*"
2. this function usage looks weird. From one hand seems it returns cuda_mem_type and is_vmm_allocation, but on other hand seems in some cases cuda_mem_type is set before calling this function. Need to have a better defined input/return values